### ISSUE TITLE
Lack of documentation, vision and plan

### ISSUE THREAD

**Issue by maximeduf** [maximeduf/multi-repo-workspace#1]
So far I have been developing mrw by myself and want to encourage contributions to this tool.

This means:
- not pushing directly to master
- create issues
- create pull requests

The first pull request that I will do revamps the documentation, add details about mrw, the wishes, the priorities, the vision. Also add the capacity to create a workspace and add a repository at a time.

So far, the master branch has been a big eggshell that deploys a cli with one command that prints hello. With this pull request coming soon, the base of the tool will be laid out and have the simplest use cases I could find that would be useful, creating workspaces and adding repositories. 

Ultimately you wouldn't want to be able to create 2 workspaces with the same name and later on, each workspace will have a database entry to keep track of workspace status and to enforce workspace dependencies. But I think being able to use mrw for all workspaces on the system to list/modify them and such would come as the 2nd PR to have basic CRUD of workspaces initialized just by entering  `mrw` to manage workspaces with the tool.

PR coming soon, comments and reactions welcome.

**Comment by jeansalama** [comment]
Good idea @maximeduf. Maybe to illustrate what is `mrw`, one can document functional use cases with desired output.
Once we will have some use cases, we will try to illustrate a minimal high level conception with diagrams.
Examples and diagrams would be very usefull to understand `mrw`.

**Comment by maximeduf** [comment]
Hey @jeansalama, I've opened a draft pull request with most of the changes I've talked about. It still needs some love but the heart of it is there. Feel free to look at the branch 'feat/workspace-and-repositories' and discuss the contents [here](https://github.com/maximeduf/multi-repo-workspace/pull/2).

The big section to look at is the [documentation](https://github.com/maximeduf/multi-repo-workspace/tree/feat/workspace-and-repositories/docs)

**Comment by jeansalama** [comment]
I began last week some reading, till now it's great. I will comment some questions in the docs PR
